os/board/rtl8721csm/src/component/common/api/wifi/rtw_wpa_supplicant/wpa_supplicant/wifi_eap_config.c

```diff
@@ -259,14 +281,18 @@ int connect_by_open_system(char *target_ssid)
 	if (target_ssid != NULL) {
 		while (1) {
 			rtw_msleep_os(500);	//wait scan complete.
-			ret = wifi_connect(target_ssid,
-							RTW_SECURITY_OPEN,
-							NULL,
-							strlen(target_ssid),
-							0,
-							0,
-							NULL);
-			if (ret == RTW_SUCCESS) {
+			wifi_manager_ap_config_s apconfig;

```

Comment:
retry_count is always 0 here, can you check how its modified and ret is not being used